pkg/controller/hawtio/hawtio_controller.go

```diff
@@ -606,6 +617,49 @@ func (r *ReconcileHawtio) Reconcile(request reconcile.Request) (reconcile.Result
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	// Reconcile console link in application menu
+	if isClusterDeployment {
+		consoleLinkCrd := &apiextensionsv1beta1.CustomResourceDefinition{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: consoleLinkCrdName}, consoleLinkCrd)
+		if err != nil && errors.IsNotFound(err) {
+			reqLogger.Info("No support for console links")
+		} else if err != nil {
+			reqLogger.Error(err, "Failed to get ConsoleLink CRD")
+		} else {
+			hawtconfig, err := osutil.GetHawtconfig(configMap)
+			if err != nil {
+				reqLogger.Error(err, "Failed to get hawtconfig")
+			} else {
+				name := instance.ObjectMeta.Name
+				consoleLink := &consolev1.ConsoleLink{}
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: name}, consoleLink)
+				if err != nil && errors.IsNotFound(err) {
+					// If not found, create a console link
+					consoleLink := osutil.NewConsoleLink(name, route, hawtconfig)
+					reqLogger.Info("Creating console link", "ConsoleLink.Name", consoleLink.Name)
+					err = r.client.Create(context.TODO(), consoleLink)
+					if err != nil {
+						reqLogger.Error(err, "Failed to create console link", "ConsoleLink.Name", consoleLink.Name)
+					}
+					// Console link created successfully - return and requeue
+					return reconcile.Result{Requeue: true}, nil
+				} else if err != nil {
+					reqLogger.Error(err, "Failed to get console link")
```

Comment:
The error should be re-enqueued. Then the else branch below can be removed.

Reply:
Done